==9313== Memcheck, a memory error detector
==9313== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==9313== Using Valgrind-3.10.0.SVN and LibVEX; rerun with -h for copyright info
==9313== Command: ./TestInteger
==9313== 
Running main() from gtest_main.cc
[==========] Running 82 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 82 tests from Integer
[ RUN      ] Integer.shift_left_digits
[       OK ] Integer.shift_left_digits (34 ms)
[ RUN      ] Integer.shift_left_digits2
[       OK ] Integer.shift_left_digits2 (10 ms)
[ RUN      ] Integer.shift_left_digits3
[       OK ] Integer.shift_left_digits3 (5 ms)
[ RUN      ] Integer.shift_left_digits4
[       OK ] Integer.shift_left_digits4 (5 ms)
[ RUN      ] Integer.shift_right_digits
[       OK ] Integer.shift_right_digits (5 ms)
[ RUN      ] Integer.shift_right_digits2
[       OK ] Integer.shift_right_digits2 (5 ms)
[ RUN      ] Integer.shift_right_digits3
[       OK ] Integer.shift_right_digits3 (5 ms)
[ RUN      ] Integer.shift_right_digits4
[       OK ] Integer.shift_right_digits4 (5 ms)
[ RUN      ] Integer.plus_digits
==9313== Invalid write of size 4
==9313==    at 0x433FBF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x409107: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c603e8 is 0 bytes after a block of size 24 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x409010: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
==9313== Invalid read of size 2
==9313==    at 0x4C2F647: memcpy@GLIBC_2.2.5 (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43EE8A: int* std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<int>(int const*, int const*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43D4D0: int* std::__copy_move_a<false, int*, int*>(int*, int*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B8CC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::__copy_move_a2<false, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438F9C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::copy<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x4334DC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x4343EC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x409107: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c603e8 is 0 bytes after a block of size 24 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x409010: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
==9313== Invalid write of size 4
==9313==    at 0x43B94D: __gnu_cxx::__enable_if<std::__is_scalar<int>::__value, void>::__type std::__fill_a<int*, int>(int*, int*, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x439090: void std::fill<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int>(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43353D: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x4343EC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x409107: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c603e8 is 0 bytes after a block of size 24 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x409010: Integer_plus_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
[       OK ] Integer.plus_digits (12 ms)
[ RUN      ] Integer.plus_digits2
==9313== Invalid write of size 4
==9313==    at 0x433FBF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x409A25: Integer_plus_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c60cec is 0 bytes after a block of size 12 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40992E: Integer_plus_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
==9313== Invalid read of size 2
==9313==    at 0x4C2F638: memcpy@GLIBC_2.2.5 (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43EE8A: int* std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<int>(int const*, int const*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43D4D0: int* std::__copy_move_a<false, int*, int*>(int*, int*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B8CC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::__copy_move_a2<false, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438F9C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::copy<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x4334DC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x4343EC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x409A25: Integer_plus_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c60cee is 2 bytes after a block of size 12 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40992E: Integer_plus_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
[       OK ] Integer.plus_digits2 (6 ms)
[ RUN      ] Integer.plus_digits3
==9313== Invalid write of size 4
==9313==    at 0x433FBF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40A343: Integer_plus_digits3_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c614cc is 0 bytes after a block of size 12 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40A24C: Integer_plus_digits3_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
[       OK ] Integer.plus_digits3 (8 ms)
[ RUN      ] Integer.plus_digits4
==9313== Invalid write of size 4
==9313==    at 0x433FBF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40AC61: Integer_plus_digits4_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c61cac is 0 bytes after a block of size 12 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40AB6A: Integer_plus_digits4_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
[       OK ] Integer.plus_digits4 (7 ms)
[ RUN      ] Integer.plus_digits5
==9313== Invalid write of size 4
==9313==    at 0x433FBF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40B5F9: Integer_plus_digits5_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c6249c is 0 bytes after a block of size 12 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40B488: Integer_plus_digits5_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
[       OK ] Integer.plus_digits5 (6 ms)
[ RUN      ] Integer.plus_digits6
==9313== Invalid write of size 4
==9313==    at 0x433FBF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40BFFA: Integer_plus_digits6_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c62c9c is 0 bytes after a block of size 12 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40BE4C: Integer_plus_digits6_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
[       OK ] Integer.plus_digits6 (7 ms)
[ RUN      ] Integer.plus_digits7
==9313== Invalid write of size 4
==9313==    at 0x433FBF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40C994: Integer_plus_digits7_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c63710 is 0 bytes after a block of size 336 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x40C860: Integer_plus_digits7_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
[       OK ] Integer.plus_digits7 (7 ms)
[ RUN      ] Integer.plus_digits8
[       OK ] Integer.plus_digits8 (6 ms)
[ RUN      ] Integer.minus_digits
[       OK ] Integer.minus_digits (26 ms)
[ RUN      ] Integer.minus_digits2
[       OK ] Integer.minus_digits2 (6 ms)
[ RUN      ] Integer.minus_digits3
[       OK ] Integer.minus_digits3 (5 ms)
[ RUN      ] Integer.minus_digits4
[       OK ] Integer.minus_digits4 (6 ms)
[ RUN      ] Integer.minus_digits5
[       OK ] Integer.minus_digits5 (6 ms)
[ RUN      ] Integer.minus_digits6
[       OK ] Integer.minus_digits6 (6 ms)
[ RUN      ] Integer.multiplies_digits
==9313== Invalid write of size 8
==9313==    at 0x4C2F5F3: memcpy@GLIBC_2.2.5 (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43EE8A: int* std::__copy_move<false, true, std::random_access_iterator_tag>::__copy_m<int>(int const*, int const*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43D4D0: int* std::__copy_move_a<false, int*, int*>(int*, int*, int*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B8CC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::__copy_move_a2<false, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438F9C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > std::copy<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x4334DC: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x434ECF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411437: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c6b7d0 is 0 bytes inside a block of size 4 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x434D56: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411437: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
going into plus digits and total.begin is 0and bottomshifted .begin is 1 and running total begin is 0
==9313== Invalid write of size 4
==9313==    at 0x439DDE: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43526B: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411437: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c6b874 is 0 bytes after a block of size 4 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x434E22: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411437: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
==9313== Invalid write of size 4
==9313==    at 0x43B94D: __gnu_cxx::__enable_if<std::__is_scalar<int>::__value, void>::__type std::__fill_a<int*, int>(int*, int*, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x439090: void std::fill<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int>(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43353D: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43A20B: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43526B: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411437: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c6b874 is 0 bytes after a block of size 4 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x434E22: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411437: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
i made it out of plus digits
==9313== Invalid write of size 4
==9313==    at 0x43B94D: __gnu_cxx::__enable_if<std::__is_scalar<int>::__value, void>::__type std::__fill_a<int*, int>(int*, int*, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x439090: void std::fill<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int>(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43353D: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > shift_left_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x434ECF: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411437: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c6b918 is 4 bytes after a block of size 4 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x434D56: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411437: Integer_multiplies_digits_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
going into plus digits and total.begin is 1and bottomshifted .begin is 0 and running total begin is 0
i made it out of plus digits
going into plus digits and total.begin is 1and bottomshifted .begin is 0 and running total begin is 0
i made it out of plus digits
TestInteger.c++:224: Failure
Value of: equal(z.begin(), z.end(), a.begin())
  Actual: false
Expected: true
[  FAILED  ] Integer.multiplies_digits (44 ms)
[ RUN      ] Integer.multiplies_digits2
going into plus digits and total.begin is 0and bottomshifted .begin is 0 and running total begin is 0
==9313== Invalid write of size 4
==9313==    at 0x439DDE: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43526B: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411D55: Integer_multiplies_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c6e2d4 is 0 bytes after a block of size 4 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x434E22: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x411D55: Integer_multiplies_digits2_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
i made it out of plus digits
going into plus digits and total.begin is 0and bottomshifted .begin is 0 and running total begin is 0
i made it out of plus digits
going into plus digits and total.begin is 0and bottomshifted .begin is 0 and running total begin is 0
i made it out of plus digits
going into plus digits and total.begin is 0and bottomshifted .begin is 0 and running total begin is 0
i made it out of plus digits
going into plus digits and total.begin is 0and bottomshifted .begin is 0 and running total begin is 0
i made it out of plus digits
going into plus digits and total.begin is 0and bottomshifted .begin is 0 and running total begin is 0
i made it out of plus digits
[       OK ] Integer.multiplies_digits2 (7 ms)
[ RUN      ] Integer.multiplies_digits3
going into plus digits and total.begin is 0and bottomshifted .begin is 1 and running total begin is 0
==9313== Invalid write of size 4
==9313==    at 0x439DDE: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > plus_digits<__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43526B: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x412673: Integer_multiplies_digits3_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x5c6f1d4 is 0 bytes after a block of size 4 alloc'd
==9313==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==9313==    by 0x43D399: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B666: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x43B766: std::_Vector_base<int, std::allocator<int> >::_M_create_storage(unsigned long) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x438E1E: std::_Vector_base<int, std::allocator<int> >::_Vector_base(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x433387: std::vector<int, std::allocator<int> >::vector(unsigned long, std::allocator<int> const&) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x434E22: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x412673: Integer_multiplies_digits3_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313== 
i made it out of plus digits
going into plus digits and total.begin is 1and bottomshifted .begin is 1 and running total begin is 0
i made it out of plus digits
going into plus digits and total.begin is 2and bottomshifted .begin is 0 and running total begin is 0
i made it out of plus digits
TestInteger.c++:240: Failure
Value of: equal(z.begin(), z.end(), a.begin())
  Actual: false
Expected: true
[  FAILED  ] Integer.multiplies_digits3 (8 ms)
[ RUN      ] Integer.multiplies_digits4
==9313== Invalid write of size 4
==9313==    at 0x43566C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x412F91: Integer_multiplies_digits4_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==9313== 
==9313== 
==9313== Process terminating with default action of signal 11 (SIGSEGV)
==9313==  Access not within mapped region at address 0x0
==9313==    at 0x43566C: __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > multiplies_digits<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > > >(__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > >, __gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x412F91: Integer_multiplies_digits4_Test::TestBody() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46D862: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46027C: testing::Test::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460313: testing::TestInfo::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x460414: testing::TestCase::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46068C: testing::internal::UnitTestImpl::RunAllTests() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x46096D: testing::UnitTest::Run() (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==    by 0x404F7F: main (in /v/filer4b/v38q001/sa4265/GP/cs378-integer/TestInteger)
==9313==  If you believe this happened as a result of a stack
==9313==  overflow in your program's main thread (unlikely but
==9313==  possible), you can try to increase the size of the
==9313==  main thread stack using the --main-stacksize= flag.
==9313==  The main thread stack size used in this run was 16777216.
==9313== 
==9313== HEAP SUMMARY:
==9313==     in use at exit: 24,271 bytes in 363 blocks
==9313==   total heap usage: 1,441 allocs, 1,078 frees, 127,083 bytes allocated
==9313== 
==9313== LEAK SUMMARY:
==9313==    definitely lost: 0 bytes in 0 blocks
==9313==    indirectly lost: 0 bytes in 0 blocks
==9313==      possibly lost: 6,483 bytes in 177 blocks
==9313==    still reachable: 17,788 bytes in 186 blocks
==9313==         suppressed: 0 bytes in 0 blocks
==9313== Rerun with --leak-check=full to see details of leaked memory
==9313== 
==9313== For counts of detected and suppressed errors, rerun with: -v
==9313== ERROR SUMMARY: 110 errors from 17 contexts (suppressed: 0 from 0)
Segmentation fault (core dumped)
